rules-java/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/UnzipArchiveToOutputFolder.java

```diff
@@ -28,196 +28,207 @@
+				.toString());
+		// mark the path to the archive
+		archiveModel.setUnzippedDirectory(newFileModel);
+		newFileModel.setParentArchive(archiveModel);
+
+		// add all unzipped files, and make sure their parent archive is set
+		recurseAndAddFiles(context, tempFolder, fileService, archiveModel,
+				newFileModel);
+	}
+
+	/**
+	 * Recurses the given folder and adds references to these files to the graph
+	 * as FileModels.
+	 *
+	 * We don't set the parent file model in the case of the inital children, as
+	 * the direct parent is really the archive itself. For example for file
+	 * "root.zip/pom.xml" - the parent for pom.xml is root.zip, not the
+	 * directory temporary directory that happens to hold it.
+	 */
+	private void recurseAndAddFiles(GraphContext context, Path tempFolder,
+			FileModelService fileService, ArchiveModel archiveModel,
+			FileModel parentFileModel) {
+		File fileReference = parentFileModel.asFile();
+		WindupJavaConfigurationService windupJavaConfigurationService = new WindupJavaConfigurationService(
+				context);
+		if (fileReference.isDirectory()) {
+			File[] subFiles = fileReference.listFiles();
+			if (subFiles != null) {
+				for (File subFile : subFiles) {
+					FileModel subFileModel = fileService.createByFilePath(
+							parentFileModel, subFile.getAbsolutePath());
+					subFileModel.setParentArchive(archiveModel);
+					// check if this file should not be ignored
+					if (checkIfIgnored(subFile,
+							windupJavaConfigurationService
+									.getIgnoredFileRegexes())) {
+						GraphService.addTypeToModel(context, subFileModel,
+								IgnoredFileModel.class);
+						LOG.info("File/Directory placed in "
+								+ subFile.getAbsolutePath()
+								+ " was ignored, because matched some of the ignore regex.");
+						continue;
+					}
+
+					if (subFile.isFile()
+							&& ZipUtil.endsWithZipExtension(subFileModel
+									.getFilePath())) {
+
+						File newZipFile = subFileModel.asFile();
+						ArchiveModel newArchiveModel = GraphService
+								.addTypeToModel(context, subFileModel,
+										ArchiveModel.class);
+						newArchiveModel.setParentArchive(archiveModel);
+						newArchiveModel.setArchiveName(newZipFile.getName());
+						archiveModel.addChildArchive(newArchiveModel);
+						unzipToTempDirectory(context, tempFolder, newZipFile,
+								newArchiveModel);
+
+					}
+
+					if (subFile.isDirectory()) {
+						recurseAndAddFiles(context, tempFolder, fileService,
+								archiveModel, subFileModel);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Checks if the file is ignored by any of the regexes specified. Full path name is taken into the account.
+	 * @param file
+	 * @param regexes
+	 * @return
+	 */
+	private boolean checkIfIgnored(File file, List<String> regexes) {
+		boolean ignored = false;
+		if (regexes != null && regexes.size() != 0) {
+			for (String r : regexes) {
+				if (Pattern.matches(r, file.getAbsolutePath())) {
```

Comment:
Need a Try/Catch/Log around this. If the expression is invalid it could crash the system.